src/menuFactory.js

```diff
@@ -22,8 +23,25 @@ export default (styles) => {
       // Order important: handle wrappers before setting sidebar state.
       this.applyWrapperStyles();
 
-      this.setState({ isOpen: !this.state.isOpen });
+      this.toggleIsOpen();
     },
+	  
+	toggleIsOpen() {
+	  if (this.props.isOpen && (typeof this.props.isOpen.value === 'boolean') && (typeof this.props.isOpen.requestChange === 'function')) {
+		this.props.isOpen.requestChange(!this.props.isOpen.value);
```

Comment:
Do you need to detect the next state with a callback function? I'm wondering if it's enough to simply pass a boolean in the props, and keep the standard toggle behaviour otherwise.